src/ocrd_models/ocrd_mets.py

```diff
@@ -608,48 +607,60 @@ def get_physical_pages(self, for_fileIds : Optional[str] = None, for_pageIds : O
+                        elif isinstance(pat, typing.Pattern):
+                            attr = next(a for a in list(METS_PAGE_DIV_ATTRIBUTE) for v in self._page_cache[a] if pat.fullmatch(v))
+                            cache_keys = [v for v in self._page_cache[attr] if pat.fullmatch(v)]
+                        else:
+                            raise ValueError
+                        if return_divs:
+                            ret += [self._page_cache[attr][v] for v in cache_keys]
+                        else:
+                            ret += cache_keys
+                except StopIteration:
+                    log.debug(f"No pattern matches any keys of any of the _page_caches. patterns: {page_attr_patterns}")
+            else:
+                while page_attr_patterns:
+                    pat = page_attr_patterns.pop(0)
                     for page in self._tree.getroot().xpath(
                             'mets:structMap[@TYPE="PHYSICAL"]/mets:div[@TYPE="physSequence"]/mets:div[@TYPE="page"]',
                             namespaces=NS):
                         try:
-                            if not attr:
-                                attr = next(a for a in METS_PAGE_DIV_ATTRIBUTE if \
-                                    page.get(a.name) in page_attr_patterns or \
-                                    any([isinstance(p, typing.Pattern) and p.fullmatch(page.get(a.name)) for p in page_attr_patterns]))
-                            attr_val = page.get(attr.name)
-                            if attr_val in page_attr_patterns or \
-                                    any([isinstance(p, typing.Pattern) and p.fullmatch(attr_val) for p in page_attr_patterns]):
-                                if return_divs:
-                                    ret.append(page)
-                                else:
-                                    ret.append(attr_val)
+                            if isinstance(pat, str):
+                                attr = next(a for a in list(METS_PAGE_DIV_ATTRIBUTE) if pat == page.get(a.name))
+                                ret.append(page if return_divs else pat)
+                            elif isinstance(pat, list):
+                                if not isinstance(pat[0], METS_PAGE_DIV_ATTRIBUTE):
+                                    pat.insert(0, next(a for a in list(METS_PAGE_DIV_ATTRIBUTE) if pat[0] == page.get(a.name)))
+                                attr_val = page.get(pat[0].name)
+                                if attr_val in pat:
+                                    pat.remove(attr_val)
+                                    ret.append(page if return_divs else attr_val)
+                            elif isinstance(pat, typing.Pattern):
+                                attr = next(a for a in list(METS_PAGE_DIV_ATTRIBUTE) if pat.fullmatch(page.get(a.name) or ''))
+                                ret.append(page if return_divs else page.get(attr.name))
```

Comment:
again, assuming we always want physical page IDs returned:
```suggestion
                                ret.append(page if return_divs else page.get('ID'))
```